src/sage/schemes/elliptic_curves/ell_curve_isogeny.py

```diff
@@ -867,9 +867,9 @@ class EllipticCurveIsogeny(EllipticCurveHom):
 
         sage: E = EllipticCurve(j=GF(7)(0))
         sage: phi = E.isogeny([E(0), E((0,1)), E((0,-1))]); phi
-        Isogeny of degree 3
-         from Elliptic Curve defined by y^2 = x^3 + 1 over Finite Field of size 7
-           to Elliptic Curve defined by y^2 = x^3 + 1 over Finite Field of size 7
+        Composite morphism of degree 3 = 3:
```

Comment:
Although this is fine, and obviously works when it's composite, I wonder if we should have a check in the format which does:

```
Composite morphism of degree 3
```

When it's prime and this format when it's composite? This would be a modification of `_repr_` for the composite morphism... 

```
Composite morphism of degree 9 = 3^2:
```